### PR TITLE
Skip statefulset scaling should happen in predictable order test

### LINKED ISSUE #47378: Test Failure: Scaling should happen in predictable order and halt if any stateful pod is unhealthy
Scaling should happen in predictable order and halt if any stateful pod is unhealthy"

Has been failing forever - possibly checking deprecated behavior?

[jenkins](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-cvm-1-6-cvm-1-7-upgrade-cluster/7#k8sio-statefulset-k8sio-basic-statefulset-functionality-statefulsetbasic-scaling-should-happen-in-predictable-order-and-halt-if-any-stateful-pod-is-unhealthy)

[test-grid](https://k8s-testgrid.appspot.com/1.6-1.7-upgrade#gke-cvm-upgrade-cluster&include-filter-by-regex=%5EOverall%24%7C%5C%5Bk8s%5C.io%5C%5D%5C%20StatefulSet%5C%20%5C%5Bk8s%5C.io%5C%5D%5C%20Basic%5C%20StatefulSet%5C%20functionality%5C%20%5C%5BStatefulSetBasic%5C%5D%5C%20Scaling%5C%20should%5C%20happen%5C%20in%5C%20predictable%5C%20order%5C%20and%5C%20halt%5C%20if%5C%20any%5C%20stateful%5C%20pod%5C%20is%5C%20unhealthy)

/assign @lavalamp 
/sig scalability

### PR DESCRIPTION
**Pull request by crimsonfaith91**
**What this PR does / why we need it**: This PR skips v1.6 test named "Scaling should happen in predictable order and halt if any stateful pod is unhealthy" to ensure the 1.6-1.7 upgrade test passes.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #47378

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: test/e2e/statefulset.go [hunk 1]
```diff
@@ -343,6 +344,8 @@ var _ = framework.KubeDescribe("StatefulSet", func() {
 		})
 
 		It("Scaling should happen in predictable order and halt if any stateful pod is unhealthy", func() {
+			framework.SkipUnlessServerVersionGTE(version.MustParseSemantic("v1.7"), f.ClientSet.Discovery())
```
Comment (foxish) [inline test/e2e/statefulset.go]: Isn't it usually defined as `v1.7.0`?

Comment (crimsonfaith91) [inline test/e2e/statefulset.go]: noted

Comment (crimsonfaith91) [inline test/e2e/statefulset.go]: ya, i will change to `v1.7.0`.

**Comment by crimsonfaith91** [comment]
@dchen1107 this PR requires `cherrypick-approved` label.

**Comment by foxish** [comment]
@kow3ns PTAL to see if we want to gate the e2e in this manner.

File: test/e2e/statefulset.go [hunk 2]
```diff
@@ -343,6 +344,8 @@ var _ = framework.KubeDescribe("StatefulSet", func() {
 		})
 
 		It("Scaling should happen in predictable order and halt if any stateful pod is unhealthy", func() {
+			framework.SkipUnlessServerVersionGTE(version.MustParseSemantic("v1.7.0"), f.ClientSet.Discovery())
```
Comment (crimsonfaith91) [inline test/e2e/statefulset.go]: This should be `SkipUnlessServerVersionLT`

**Pull request merged**